In [6]:
import pandas as pd

features_df = pd.read_csv(r"C:\Users\Meeting\nova\data-pipeline\processed_data\features_20260921.csv", parse_dates=["date"])
print(f"Loaded {len(features_df)} rows, {len(features_df.columns)} columns")

Loaded 2505 rows, 28 columns


In [7]:
print("RSI range:", features_df['rsi_14'].min(), "to", features_df['rsi_14'].max())
print("\nSample of AAPL features (most recent 5 rows):")
aapl = features_df[features_df['symbol'] == 'AAPL'].tail(5)
print(aapl[['date', 'close', 'ma_7', 'ma_30', 'rsi_14', 'macd', 'bb_upper', 'bb_lower']])

RSI range: 6.467188262277304 to 97.54918875734111

Sample of AAPL features (most recent 5 rows):
          date       close        ma_7       ma_30     rsi_14      macd  \
496 2026-09-14  333.079987  324.522853  314.482826  70.523448  3.970020   
497 2026-09-15  331.339996  324.969997  315.422207  68.906524  4.338051   
498 2026-09-16  332.410004  326.747140  316.198760  67.484315  4.662314   
499 2026-09-17  337.000000  329.715711  317.074359  69.434817  5.229389   
500 2026-09-18  336.130005  332.685713  317.873999  65.375253  5.544682   

       bb_upper    bb_lower  
496  334.020515  301.946480  
497  335.253075  303.288920  
498  336.776587  304.003409  
499  339.277322  303.519676  
500  341.007931  304.272068  


## Feature Verification

- Engineered 28 features: returns (1/7/14-day), moving averages (SMA 7/30, EMA 12/26), 
  volatility, RSI, MACD, Bollinger Bands, and lag features
- RSI correctly bounded between 0-100 (observed range: 6.5 to 97.5)
- AAPL sample shows RSI ~65-70 (near overbought) with price tracking near upper Bollinger Band ($336-341)
- Values are consistent and sane — ready for use in forecasting models (Phase 5+)